app/models/admin_only/file_delivery_method.rb

```diff
@@ -14,13 +14,30 @@ class FileDeliveryMethod < ApplicationRecord
 
     validates :description_en, :description_sv, presence: true
 
-    validates :name, presence: true,
+    validates :name, presence: true, uniqueness: true,
```

Comment:
This (and the `_spec`) made me think and read about `validates uniqueness` more:

From [the Rails Guide about *validates uniqueness*:](https://guides.rubyonrails.org/active_record_validations.html#uniqueness)
>  It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

so: 
- We need to also create a unique index in the migration so the DB ensures uniqueness, too:
```
add_index :file_delivery_methods, [:name], unique: true
```
_(I think that code is right.  I'm winging it, so don't count on it.)_

  And yes, we should then be doing that _everywhere_ that we have a uniqueness constraint

- We should specify that a nil/NULL value is not acceptable in both the migration and the model.  Yes?

Reply:
I added the index, and the restriction on :null to the migration.

The model validates the presence of `name`, so that covers `nil` values.

Reply:
If this were a bigger, more complex system, I'd push for a db statement ` ..null: false ...` in the migration so that we also have that guard in the db.   But... I think it'll be just fine  as you have it now. (with the db index)